### PR TITLE
support windows \r encoding

### PR DESCRIPTION
**Pull request by LemonHX**
support windows benchmarking

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

**Comment by LemonHX** [comment]
@wrwg transfered

File: language/move-command-line-common/src/character_sets.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ pub fn is_permitted_printable_char(c: char) -> bool {
 /// The only permitted newline character is \n. All others are invalid.
 pub fn is_permitted_newline_char(c: char) -> bool {
     let x = c as u32;
-    x == 0x0A
+    x == 0x0A || x == 0x0D
```
Comment (tnowacki) [inline language/move-command-line-common/src/character_sets.rs:21]: Thanks for the PR! And sorry you are having issues, but we won't be able to change this right away.
This restriction was made in response to a potential security vulnerability where comments in Move code commented out the following line, without the programmer knowing. 

We had discussed relaxing this late last year, but did not make any firm plans. We could file an issue and will need to discuss this further

**Review by tnowacki — CHANGES_REQUESTED** [review]

**Comment by tnowacki** [comment]
After discussing it, we will look at a PR for unicode characters in comments, but will not be accepting anything at this time that allows `\r`, sorry

**Comment by wrwg** [comment]
Just curious: if only the specific Windows sequence \r\n would be accepted, but not a standalone \r, wouldn't that solve the security problem? It appears pretty restrictive to not allow Move on native Windows.

**Comment by sblackshear** [comment]
T

> Just curious: if only the specific Windows sequence \r\n would be accepted, but not a standalone \r, wouldn't that solve the security problem? It appears pretty restrictive to not allow Move on native Windows.

That solution sounds reasonable to me, but I don't have the right context on the parser to understand how easy it is to do this.

@LemonHX : would you mind de-scoping this PR to the parser change only (i.e., move the benchmarking code to a separate PR?)

**Comment by LemonHX** [comment]
@sblackshear I will do that

**Comment by LemonHX** [comment]
@wrwg so we could add to our documentation that if you are windows user please config your git before generating any move file

**Comment by LemonHX** [comment]
@wrwg \r is used in native mac...

**Comment by wrwg** [comment]
> @wrwg \r is used in native mac...

... and we will not allow it if it is used. We only allow regular ascii files, where lines can be separated either with \n (mac) or \r\n (windows). I think this is very easy to add to the lexer.

**Comment by wrwg** [comment]
Superseded by #90

**Pull request closed without merging**